lib/ts/recipe/thirdparty/providers/custom.ts

```diff
@@ -335,6 +335,19 @@ export default function NewProvider(input: ProviderInput): TypeProvider {
                 rawUserInfoFromProvider.fromUserInfoAPI = userInfoFromAccessToken;
             }
 
+            /**
+             * This is intentionally not part of the above if block. This is because the user may want to validate the access
+             * token payload even if the user info API has not been provided by the provider. In this case they would get an
+             * empty object and they can fail if they always expect a non-empty object.
+             */
+            if (impl.config.validateAccessToken !== undefined) {
+                await impl.config.validateAccessToken({
+                    accessToken: accessToken,
+                    clientConfig: impl.config,
+                    userContext,
+                });
+            }
```

Comment:
this should happen before the above if and we need to check if the access token is undefined or not